### PR TITLE
feat: add focus and blur methods by reference

### PR DESCRIPTION
**Pull request by cervisebas**
I would like the `focus()` and `blur()` methods to be added by reference, so that, through external components, these behaviors could be controlled.

Code example:
```typescript
import { useRef, useState } from 'react';
import { View } from 'react-native';
import { Dropdown, DropdownRef } from 'react-native-paper-dropdown';
import { Button } from 'react-native-paper';

const OPTIONS = [
  { label: 'Male', value: 'male' },
  { label: 'Female', value: 'female' },
  { label: 'Other', value: 'other' },
];

export default function App() {
  const [gender, setGender] = useState<string>();
  const refDropdown = useRef<DropdownRef>(null);

  return(<View style={{ flexDirection: 'column' }}>
    <Dropdown
      ref={refDropdown}
      label={'Gender'}
      placeholder="Select Gender"
      options={OPTIONS}
      value={gender}
      onSelect={setGender}
    />
    
    <Button
      mode={'contained'}
      onPress={() => refDropdown.current?.focus()}
    >
      Focus
    </Button>
  </View>);
}
```

Video example:

https://github.com/user-attachments/assets/6fb7e010-e4d2-4a32-8205-d0913501f65d

## Summary by CodeRabbit

- **New Features**
  - Introduced a dropdown component that can be focused and blurred via external buttons, enhancing user interaction.
  - Added a multi-select dropdown with similar focus and blur capabilities for improved usability.
  - Implemented error handling in the DropdownInput component to provide user feedback on form inputs.
  
- **Improvements**
  - Enhanced the Dropdown and MultiSelectDropdown components to support imperative method calls, allowing better integration within forms and other interactive elements.

- **Documentation**
  - Updated the documentation to include new methods for controlling dropdown visibility (`focus()` and `blur()`).
  - Added a new type definition for DropdownRef, providing clear structure for dropdown control methods.

### CODE REVIEW

**Comment by fateh999** [comment]
@cervisebas thanks for the pr, it is a good addition can you resolve the conflicts so I can merge it, I fixed some points.

**Comment by fateh999** [comment]
@cervisebas once it gets merged we should update the documentation as well.

File: src/index.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import Dropdown from './dropdown';
+import Dropdown from './DropDown';
```
Comment (fateh999) [inline src/index.tsx:1]: I have fixed the capitalisation issue of this file name, it will be dropdown.tsx

**Comment by cervisebas** [comment]
> @cervisebas thanks for the pr, it is a good addition can you resolve the conflicts so I can merge it, I fixed some points.

Where can I see the conflicts?

**Comment by fateh999** [comment]
> > @cervisebas thanks for the pr, it is a good addition can you resolve the conflicts so I can merge it, I fixed some points.
> 
> Where can I see the conflicts?

can you merge latest main in your branch. It should give you conflicts

**Comment by cervisebas** [comment]
@fateh999 Merge completed!

**Comment by cervisebas** [comment]
> > > @cervisebas thanks for the pr, it is a good addition can you resolve the conflicts so I can merge it, I fixed some points.
> > 
> > 
> > Where can I see the conflicts?
> 
> can you merge latest main in your branch. It should give you conflicts

Thanks for the help

**Review by fateh999 — APPROVED** [review]

**Pull request merged**